File: backend/src/main/java/com/grouply/backend/project_member/dto/ProjectMemberDTO.java
package com.grouply.backend.project_member.dto;
import com.grouply.backend.project_member.ProjectPosition;
import com.grouply.backend.project_member.ProjectRole;
import com.grouply.backend.user.Dtos.UserDTO;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProjectMemberDTO {
    private Long id;
    private UserDTO user;
    private ProjectPosition projectPosition;
    private ProjectRole projectRole;
}

File: frontend/src/dtos/models_dtos/project_member_dto.ts

import type { ProjectPosition } from "../enums/ProjectPosition"
import type { ProjectRole } from "../enums/ProjectRole"
import type { UserDTO } from "./user_dto"

export class ProjectMemberDTO {
    id: number
    user: UserDTO
    projectPosition: ProjectPosition
    projectRole: ProjectRole

    constructor(id: number, user: UserDTO, projectPosition: ProjectPosition, projectRole: ProjectRole) {
        this.id = id
        this.user = user
        this.projectPosition = projectPosition
        this.projectRole = projectRole
    }
}